src/elfpy/pricing_models/base.py

```diff
@@ -568,14 +571,6 @@ def check_input_assertions(
             "pricing_models.check_input_assertions: ERROR: "
             f"expected quantity.amount >= {WEI}, not {quantity.amount}!"
         )
-        assert market_state.share_reserves >= WEI, (
```

Comment:
these seem like good checks to have no?

Reply:
yeah, but they don't work in a world where we end up with negative reserves, then catch them in `apply_deltas` and set reserves to 0 when the negative value is within a precision threshold. I guess we could set that amount to 1 WEI instead of 0? that way it would be closer to solidity, which is probably going to keep this check?

Reply:
We really should have these rounding conditionals & checks located in one place. It's dangerous to sprinkle them throughout the code -- you can get values transforming that you don't expect. Debugging becomes increasingly difficult. I _think_ that is what Mihai is suggesting, but I just want to make it clear from my end.

Reply:
But on the other hand we need to have full coverage with these model assertions.